src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -51,56 +50,55 @@
 
 namespace
 {
-    TriStateBool jsonValueToTriStateBool(const QJsonValue &jsonVal)
+    std::optional<bool> jsonValueToOptionalBool(const QJsonValue &jsonVal)
     {
         if (jsonVal.isBool())
-            return TriStateBool(jsonVal.toBool());
+            return jsonVal.toBool();
 
         if (!jsonVal.isNull())
             qDebug() << Q_FUNC_INFO << "Incorrect value" << jsonVal.toVariant();
 
-        return TriStateBool::Undefined;
+        return std::nullopt;
     }
 
-    QJsonValue triStateBoolToJsonValue(const TriStateBool triStateBool)
+    QJsonValue optionalBoolToJsonValue(const std::optional<bool> boolValue)
     {
-        switch (static_cast<signed char>(triStateBool))
-        {
-        case 0:  return false;
-        case 1:  return true;
-        default: return {};
-        }
+        if (!boolValue.has_value())
+            return {};
+
+        return *boolValue;
```

Comment:
this can be a single line if you want: 
```c++
return boolValue.has_value() ? *boolValue : {};
// or maybe
return boolValue.value_or({});  // untested
```

Reply:
```c++
// or maybe
return boolValue.value_or({});  // untested
```

It won't work as expected. It returns `false` in case of empty optional.
Will use first suggestion.